### PR TITLE
feat: pull out HeaderAuthentication

### PR DESCRIPTION
**Pull request by r1b**
Test?

### CODE REVIEW

File: flask_resty/authentication.py [hunk 1]
```diff
@@ -62,3 +65,45 @@ class NoOpAuthentication:
...
+    #: used as credentials if the Authorization request header is missing.
+    credentials_arg = None
+
+    def get_request_credentials(self):
+        authorization = flask.request.headers.get("Authorization")
+
+        if authorization is None:
+            if self.credentials_arg is not None:
+                return flask.request.args.get(self.credentials_arg)
+            raise ApiError(401, {"code": "invalid_authorization.missing"})
```
Comment (taion) [inline flask_resty/authentication.py]: i think we just return None for “no credentials” here. the check for “credentials are required” is proper to authZ

File: flask_resty/authentication.py [hunk 2]
```diff
@@ -62,3 +65,45 @@ class NoOpAuthentication:
...
+
+        return self.get_credentials_from_authorization(authorization)
+
+    def get_credentials_from_authorization(self, authorization):
+        if self.header_scheme is None:
```
Comment (taion) [inline flask_resty/authentication.py]: could just default this to Bearer. the other schemes all have handshakes and challenges that were not implementing

**Comment by r1b** [comment]
Ping, I renamed this back to `HeaderAuthentication` since it can be instantiated as-is with a default scheme. Added tests, not sure why coverage is complaining.

**Comment by r1b** [comment]
Ah nvm it was because I broke JWT :P

**Pull request merged**